cirq-core/cirq/experiments/fidelity_estimation_test.py

```diff
@@ -81,7 +81,7 @@ def test_xeb_fidelity(depolarization, estimator):
         f2 = cirq.xeb_fidelity(
             circuit, bitstrings, qubits, amplitudes=amplitudes, estimator=estimator
         )
-        assert np.abs(f - f2) < 1e-6
+        assert np.abs(f - f2) < 2e-6
```

Comment:
lets use `pytest.approx` instead
```py3
assert f2 == pytest.approx(f, rel=1e-6)
```
This will assert that $f2$ equals $f$ up to relative error of 1e-6. This should be more robust unless $f$ is almost zero in which case we can make it
```py3
assert f2 == pytest.approx(f, rel=1e-6, abs=1e-6)
```
so that it works for all cases

Reply:
@NoureldinYosri - thank you for the suggestion, done in 9ddcbaabc713dfefbf5e9628cd37cc4ee0bbb803.
Good to know about `pytest.approx` it is however not used in cirq.  `numpy.allclose` does the same thing here so I opted for that one.

Reply:
relative tolerance works only for large numbers. for small number absolute tolerance is needed so
```py3
np.allclose(f, f2, atol=1e-6, rtol=1e-6)
```

Reply:
In this case the initial absolute value comparison was better as the fidelity is supposed to be on the [0, 1] interval.
For some reason the round-off errors on Mac got it over the threshold.